appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -0,0 +1,229 @@
+import * as k8s from "@kubernetes/client-node";
+import forge from "node-forge";
+import pem from "pem";
+
+class WebhookCertData {
+    caCert: string;
+    tlsCert: string;
+    tlsKey: string;
+}
+
+const hostname = process.env.HOSTNAME;
+
+export class CertificateManager {
+
+    private static makeNumberPositive = (hexString) => {
+        let mostSignificativeHexDigitAsInt = parseInt(hexString[0], 16);
+    
+        if (mostSignificativeHexDigitAsInt < 8) return hexString;
+    
+        mostSignificativeHexDigitAsInt -= 8
+        return mostSignificativeHexDigitAsInt.toString() + hexString.substring(1)
+    }
+    
+    // Generate a random serial number for the Certificate
+    private static randomSerialNumber = () => {
+        return CertificateManager.makeNumberPositive(forge.util.bytesToHex(forge.random.getBytesSync(20)));
+    }
+
+    private static async GenerateSelfSignedCertificate(): Promise<WebhookCertData> {
+
+        let caCert: forge.pki.Certificate = forge.pki.createCertificate();
+        let keys = forge.pki.rsa.generateKeyPair(4096);
+        caCert.serialNumber = CertificateManager.randomSerialNumber();
+        caCert.publicKey = keys.publicKey;
+        caCert.privateKey = keys.privateKey;
+        caCert.validity.notBefore = new Date(2023,6,20,0,0,0,0);
```

Comment:
Let's not hardcode these 2 dates, let's make it between 5 minutes ago and 2 years from now.

Reply:
This is not addressed yet. Let's not hardcode the dates.